README.md

```diff
@@ -1,45 +1,97 @@
-
-## Documents
-
-- [**Roadmap**](/ROADMAP.md)
-- **Specifications:**
-  - ![](https://img.shields.io/badge/status-draft-yellow.svg?style=flat-square) [`IPLD`](/IPLD.md) - spec about the data model, pointers and link formats
-  - ![](https://img.shields.io/badge/status-wip-orange.svg?style=flat-square) `IPLD Selectors` - spec about simple language to select multiple unknown nodes in a graph
-  - ![](https://img.shields.io/badge/status-wip-orange.svg?style=flat-square) `IPLD Transformations` - spec about the language to trasform an IPLD graph into another
-  - ![](https://img.shields.io/badge/status-reliable-green.svg?style=flat-square) [`CID (Content IDentifier)`](https://github.com/ipld/cid)
-  - ![](https://img.shields.io/badge/status-wip-orange.svg?style=flat-square) [`IPLD Formats`](https://github.com/ipld/interface-ipld-format) - interface definition for adding support to different formats
-  - ![](https://img.shields.io/badge/status-draft-yellow.svg?style=flat-square) [`CAR`](/CAR.md) - Content Addressable Archives
+IPLD is not a single specification, it is a set of specifications.
+
+```
+                        The IPLD Stack
+                                +-----------------------------+
+                +-------------+ |                             |
+                |             | | End-User Application Stacks |
+                | MFS in IPFS | |                             |
+                |             | +-----------------------------+
+                +-------------+ |                             |
+                |             | | Structured Data w/ indexes  |
+                |  unixfs v2  | |     VR, Geo, SQL, etc.      |     +----------+
+                |             | |                             |     |          |
+                +-------------+ +-----------------------------+     |  MFS in  |
+                |             | |                             |     |   IPFS   |
+                |    HAMT     | |   Sorted Index (sharded)    |     |          |
+                |             | |                             |     +----------+
+                +-------------+-+-----------------------------+     |          |
+                |                                             |     |  unixfs  |
+                |          Complex Data Structures            |     |    v1    |
+                |                                             |     |          |
++------------------------------------------------------------------------------+
+|               |                                             |     |          |
+|               |        dag-json         dag-cbor            |     |          |
+|               |                                             |     |          |
+|    Codecs     +---------------------------------------------+ git |  dag-pb  |
```

Comment:
Not trying to paint a bikeshed, but can we s/Codecs/Formats? We've been calling these formats for a while now.

Reply:
Speaking only from personal experience, I found that `Format` sent me down a path to a different understanding to that which `Codec` does. So I found the rephrasing to `Codec` useful. Just my 2 cents.

Reply:
It would be great to have some definitions and examples to illustrate the IPFS/IPLD usage of: codecs; formats; types/typing; schema. I've seen them all used in IPLD discussions but depending on your background these can have different meanings and connotations that don't exactly match IPLD/IPFS usage.

Reply:
@diasdavid the CID spec uses "Codec" consistently and that's the most consistent use I could find. I've seen format used interchangeably with codec in a few places but I've rarely seen codec refer to something *inside* a format.

If we are borrowing from prior definitions then "format" would need to be inside of "codec" as codecs traditionally compress data *into* a "format."

Reply:
I'll toss this into the ring: "Representations".

I did some research spike code in a vacuum last week and that's what I came up with.

ISTM this might fit well because it allows us to consider representations that aren't actually serial at all.  For example, when I have an in-memory `Node` interface hanging onto a bunch of data... it's in IPLD Data Model.  I can query and handle it like IPLD.  But it's not in CBOR... yet.  This in-memory is a valid "representation"; like any other representations, e.g. JSON, it has to be converted to a serializable representation (namely, CBOR) to be shared and hashed, but it's still definitely a representation of IPLD.

Reply:
@warpfork so, you're recommending "representation" for the term used to describe in-memory de-serialized nodes? 

I like it, I don't think we've had a consistent term for that yet.

Reply:
I'd actually propose "IPLD Representation" for *all of*: dag-cbor, dag-json, and go-inmem, etc.  I can traverse, etc, each of those with the standard IPLD Node interfaces.  They'll all be confined by the IPLD Data Model core types.

The first two just happen to be an IPLD *Serial* Representation.